contrib/pkg/sets/sets.go

```diff
@@ -34,18 +34,25 @@ func Key(id ezkube.ResourceId) string {
 	if id == nil {
 		return "<unknown>"
 	}
-	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
-		b.WriteString(clusterId.GetName())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetNamespace())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetClusterName())
-		return b.String()
-	}
 	b.WriteString(id.GetName())
 	b.WriteString(separator)
 	b.WriteString(id.GetNamespace())
 	b.WriteString(separator)
+	// handle the possibility that clusterName could be set either as an annotation (new way)
+	// or as a field (old way pre-k8s 1.25)
+	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
+		clusterNameByAnnotation := ezkube.GetClusterName(clusterId)
+		if clusterNameByAnnotation != "" {
+			b.WriteString(separator)
+			b.WriteString(clusterNameByAnnotation)
+		} else if deprecatedClusterId, ok := id.(interface{ GetClusterName() string }); ok {
```

Comment:
why are we definining the interface in line? also why do we have a double else if here?

Reply:
1. so we don't have to expose the deprecatedClusterId interface defined in ezkube, we can't really define it in this file since we need it for ConvertRefToId(...)
2. double else if bc there is a possibility that a pre-k8s-1.25 object either has the annotation or the clusterName field defined, so we should handle both cases (inside first if). If the object is of ClusterObjectRef type, then it will not match the new ClusterResourceId interface (first if), and thus needs to be handled as follow-on else if statement.